examples/p4est_2d_dgsem/elixir_eulermulti_shock.jl

```diff
@@ -0,0 +1,127 @@
+using OrdinaryDiffEqSSPRK, OrdinaryDiffEqLowStorageRK
+using Trixi
+
+# 1) Dry Air  2) SF_6
+equations = CompressibleEulerMulticomponentEquations2D(gammas = (1.4, 1.648),
+                                                       gas_constants = (0.287, 1.578))
+
+"""
+    initial_condition_shock(coordinates, t, equations::CompressibleEulerEquations2D)
+
+Shock traveling from left to right where it interacts with a Perturbed interface.
+"""
+@inline function initial_condition_shock(x, t,
+                                        equations::CompressibleEulerMulticomponentEquations2D)
+    rho_0 = 1.25 # kg/m^3
+    p_0 = 101325 # Pa
+    T_0 = 293 # K
+    u_0 = 352 # m/s
+    d = 20
+    w = 40
+
+    if x[1] < 25
+        # Shock region.
+        v1 = 0.35
+        v2 = 0.0
+        rho1 = 1.72
+        rho2 = 0.03
+        p = 1.57
+    elseif (x[1] <= 30) || (x[1] <= 70 && abs(125 - x[2]) > w/2)
+        # Intermediate region.
+        v1 = 0.0
+        v2 = 0.0
+        rho1 = 1.25
+        rho2 = 0.03
+        p = 1.0
+    else (x[1] <= 70 + d)
+        # SF_6 region.
+        v1 = 0.0
+        v2 = 0.0
+        rho1 = 0.03
+        rho2 = 6.03 #SF_6
+        p = 1.0
+    end
+
+    return prim2cons(SVector(v1, v2, p, rho1, rho2), equations)
+end
+
+# Define the simulation.
+initial_condition = initial_condition_shock
+
+surface_flux = flux_lax_friedrichs
+volume_flux = flux_ranocha
+basis = LobattoLegendreBasis(3)
+
+limiter_idp = SubcellLimiterIDP(equations, basis;
+                                positivity_variables_cons = ["rho" * string(i)
+                                                             for i in eachcomponent(equations)])
+
+volume_integral = VolumeIntegralSubcellLimiting(limiter_idp;
+                                                volume_flux_dg = volume_flux,
+                                                volume_flux_fv = surface_flux)
```

Comment:
Did you check this code for type instabilities? Do you get type instabilities with mainstream solvers like shock capturing Hennemann & Gassner?

Reply:
+1 that is something you should check first